src/dealer/DealerWallet.ts

```diff
@@ -0,0 +1,515 @@
+      usdPnl: balance.total.USD,
+    }
+  }
+
+  async getNextFundingRate() {
+    const {
+      result: { nextFundingRate },
+    } = await this.exchange.getNextFundingRate()
+    return nextFundingRate
+  }
+
+  async getAccountPosition() {
+    // FIXME this helper function is inverse?
+    // or because price = usd/btc, usd/sats, sat or btc are in the denominator
+    // and therefore the "inverse" make sense...?
+    const btcPrice = btc2sat(UserWallet.lastPrice)
+
+    this.logger.debug({ btcPrice }, "[DEBUG] is price set for the dealer?")
+
+    // TODO: what is being returned if no order had been placed?
+    // probably an empty array
+
+    // tmp for log
+    const result = await this.exchange.privateGetAccount()
+    this.logger.debug({ result }, "full result of this.exchange.privateGetAccount")
+
+    const {
+      result: {
+        collateral,
+        positions,
+        chargeInterestOnNegativeUsd,
+        marginFraction,
+        totalAccountValue,
+      },
+    } = result
+    this.logger.debug(
+      { collateral, positions, chargeInterestOnNegativeUsd, marginFraction },
+      "value kept from this.exchange.privateGetAccount",
+    )
+
+    // getPositions() instead ?!?
+    const positionBtcPerp = _.find(positions, { future: this.exchange.symbol })
+    this.logger.debug({ positionBtcPerp }, "positionBtcPerp result")
+
+    const {
+      netSize = 0,
+      estimatedLiquidationPrice,
+      collateralUsed,
+      maintenanceMarginRequirement,
+    } = positionBtcPerp ?? {}
+
+    // TODO: check this is the intended settings
+    assert(chargeInterestOnNegativeUsd === true)
+
+    assert(netSize <= 0)
+
+    return {
+      // making netSize positive to simplify calculation
+      // we should always be short so netSize should initially
+      // always be negative
+      btc: -netSize,
+
+      usd: -netSize * btcPrice,
+      estimatedLiquidationPrice,
+      collateralUsed, // USD
+      maintenanceMarginRequirement, // start at 0.03 but increase with position side
+
+      collateral, // in USD
+
+      totalAccountValue,
+
+      // if there is no collateral, marginFraction will be null. this is equivalent to infinite leverage.
+      leverage: marginFraction ? 1 / marginFraction : Number.POSITIVE_INFINITY,
+    }
+  }
+
+  static getExposureRatio({ usdLiability, usdExposure }) {
+    return {
+      ratio: usdExposure / usdLiability,
+      diff: usdExposure - usdLiability,
+    }
+  }
+
+  // we need to rebalance when price is increasing/decreasing.
+  // if price increase, then the short position risk being liquidated
+  // so we need to sned btc from the node to the exchange
+  // if price decrease, then there would be too much btc on the exchange
+  // the account won't be at the irsk of being liquidated in this case
+  // but then the custody risk of the exchange increases
+  static isRebalanceNeeded({ usdLiability, btcPrice, usdCollateral }) {
+    type IDepositOrWithdraw = "withdraw" | "deposit" | null
```

Comment:
Can we avoid using `| null` in our types. What exactly does `null` signify here? Does it mean `no action required`? Then make the type `type IDepositOrWithdraw = "withdraw" | "deposit" | "neither" or something like that.